### PR TITLE
Clean unused nat and middleman msg

### PR DESCRIPTION
**Pull request by ggruszczynski**
Since golem-messages has been extracted to another repo, here is a new PR.
The https://github.com/golemfactory/golem/pull/1606 and https://github.com/golemfactory/golem/pull/1609
are merged into this PR and closed.

related to: https://github.com/golemfactory/golem-messages/pull/3

### CODE REVIEW

File: golem/core/hostaddress.py [hunk 1]
```diff
@@ -112,9 +112,10 @@ def get_external_address(source_port=0):
     :return (str, int, str): tuple with host public address, public port that is
     mapped to local <source_port> and this host nat type
     """
-    nat_type, external_ip, external_port = stun.get_ip_info(source_port=source_port)
-    logger.debug("NAT {}, external [{}] {}".format(nat_type, external_ip, external_port))
-    return external_ip, external_port, nat_type
+    _, external_ip, external_port = stun.get_ip_info(source_port=source_port)
+    logger.debug("External_ip [{}] External_port {}".
+        format(external_ip, external_port))
```
Comment (jiivan) [inline golem/core/hostaddress.py]: Maybe leave logging of`nat_type` in logger but just don't return it?

File: golem/network/p2p/node.py [hunk 2]
```diff
@@ -32,16 +31,15 @@ def __init__(self,
...
         self.prv_addresses = []  # type: List[str]
 
-        self.nat_type = nat_type
         self.port_status = None
 
     def collect_network_info(self, seed_host=None, use_ipv6=False):
         if not self.pub_addr:
             if self.prv_port:
-                self.pub_addr, self.pub_port, self.nat_type = \
+                self.pub_addr, self.pub_port, _ = \
```
Comment (jiivan) [inline golem/network/p2p/node.py]: Last parameter (`nat_type`) is removed right?

**Review by jiivan — CHANGES_REQUESTED** [review]
Great job! Only small fixes needed

File: golem/core/hostaddress.py [hunk 3]
```diff
@@ -112,9 +112,10 @@ def get_external_address(source_port=0):
     :return (str, int, str): tuple with host public address, public port that is
     mapped to local <source_port> and this host nat type
     """
-    nat_type, external_ip, external_port = stun.get_ip_info(source_port=source_port)
-    logger.debug("NAT {}, external [{}] {}".format(nat_type, external_ip, external_port))
-    return external_ip, external_port, nat_type
+    _, external_ip, external_port = stun.get_ip_info(source_port=source_port)
+    logger.debug("External_ip [{}] External_port {}".
```
Comment (maaktweluit) [inline golem/core/hostaddress.py]: maybe use the native logger formatter, this way the string will not be calculated if the level is off.
```
logger.debug("%r", value)
```

Comment (ggruszczynski) [reply golem/core/hostaddress.py]: done

File: golem/network/p2p/node.py (hunk 2 above)
Comment (ggruszczynski) [reply golem/network/p2p/node.py]: yes, the extra `_` is also removed.

File: golem/core/hostaddress.py (hunk 1 above)
Comment (ggruszczynski) [reply golem/core/hostaddress.py]: done

File: golem/core/hostaddress.py [hunk 4]
```diff
@@ -113,9 +113,9 @@ def get_external_address(source_port=0):
     mapped to local <source_port> and this host nat type
     """
     nat_type, external_ip, external_port = stun.get_ip_info(source_port=source_port)
-    logger.debug("NAT {}, external [{}] {}".format(nat_type, external_ip, external_port))
-    return external_ip, external_port, nat_type
-
+    logger.debug("NAT %r, external_ip [%r] External_port %r",
+                 nat_type, external_ip, external_port)
+    return external_ip, external_port
```
Comment (etam) [inline golem/core/hostaddress.py]: update documentation about returned values

**Review by jiivan — APPROVED** [review]

**Review by etam — CHANGES_REQUESTED** [review]

File: golem/core/hostaddress.py (hunk 4 above)
Comment (ggruszczynski) [reply golem/core/hostaddress.py]: done

**Review by etam — APPROVED** [review]

File: golem/network/p2p/node.py [hunk 5]
```diff
@@ -32,16 +31,15 @@ def __init__(self,
         self.pub_addr = pub_addr
         self.prv_addresses = []  # type: List[str]
 
-        self.nat_type = nat_type
```
Comment (mfranciszkiewicz) [inline golem/network/p2p/node.py:35]: Please do not remove this property, it's still useful for stats / debugging connectivity.

Comment (ggruszczynski) [reply golem/network/p2p/node.py:35]: ok, self.nat_type reverted.

**Pull request merged**